src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -13,12 +13,19 @@
 
 namespace Sylius\Component\Core\Calculator;
 
+use Sylius\Component\Core\Checker\ProductVariantLowestPriceDisplayCheckerInterface;
 use Sylius\Component\Core\Exception\MissingChannelConfigurationException;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Webmozart\Assert\Assert;
 
 final class ProductVariantPriceCalculator implements ProductVariantPricesCalculatorInterface
 {
+    public function __construct(private ProductVariantLowestPriceDisplayCheckerInterface $productVariantLowestPriceDisplayChecker)
```

Comment:
Set it to null by default and trigger a deprecation for not passing this argument with a note in UPGRADE file